### PR TITLE
[INLONG-10986][SDK] Transform support REGEXP_MATCHES(str, pattern) function

### LINKED ISSUE #10986: [Feature][SDK] Transform support REGEXP_MATCHES(str, pattern) function
### Description

parent: #10796 
* REGEXP_MATCHES(str, pattern): a powerful function in PostgreSQL that matches regular expression patterns in strings and returns all matched results

### Use case

* Matching a single pattern: 
Let's say we have a string from which we want to extract substrings that match a particular pattern
REGEXP_MATCHES('The quick brown fox', 'quick')--returns {quick}
* Extract multiple subpatterns
Multiple substrings can be extracted using capture groups in regular expressions
REGEXP_MATCHES('User: Alice, ID: 12345', 'User: (\w+), ID: (\d+)')--returns {Alice,12345}
* Globally match multiple patterns
The g flag can be used when you want to match all the substrings that occur
REGEXP_MATCHES('foo 123 bar 456', '\d+', 'g')--returns {"123","456"}
* Case matching is ignored
Use the i flag to ignore case for matching
REGEXP_MATCHES('Hello World', 'hello', 'i')--returns {Hello}
* Multi-line pattern matching
Using the m flag allows regular expressions to match across multiple lines
REGEXP_MATCHES('First line\nSecond line', '^Second', 'm')--returns {Second}
* Some Announcements：
REGEXP_MATCHES() returns an array, so be careful about using arrays when processing the results.
If the global flag g is not used, only the first matching substring is returned.

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by emptyOVO**
Fixes #10986 

### Motivation
add RegexpMatchesFunction class to support postgresql function REGEXP_MATCHES(str, pattern), provide unit tests

### Verifying this change
Matching a single pattern:
Let's say we have a string from which we want to extract substrings that match a particular pattern
* REGEXP_MATCHES('The quick brown fox', 'quick')--returns {quick}
Extract multiple subpatterns
Multiple substrings can be extracted using capture groups in regular expressions
* REGEXP_MATCHES('User: Alice, ID: 12345', 'User: (\w+), ID: (\d+)')--returns {Alice,12345}
Globally match multiple patterns
The g flag can be used when you want to match all the substrings that occur
* REGEXP_MATCHES('foo 123 bar 456', '\d+', 'g')--returns {"123","456"}
Case matching is ignored
Use the i flag to ignore case for matching
* REGEXP_MATCHES('Hello World', 'hello', 'i')--returns {Hello}
Multi-line pattern matching
Using the m flag allows regular expressions to match across multiple lines
* REGEXP_MATCHES('First line\nSecond line', '^Second', 'm')--returns {Second}
Some Announcements：
REGEXP_MATCHES() returns an array, so be careful about using arrays when processing the results.
If the global flag g is not used, only the first matching substring is returned.
If there are multiple flags, the effects need to be performed simultaneously
*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/RegexpMatchesFunction.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@TransformFunction(names = {"regexp_matches"})
+public class RegexpMatchesFunction implements ValueParser {
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/RegexpMatchesFunction.java:36]: Add descriptions

Comment (emptyOVO) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/RegexpMatchesFunction.java:47]: done

**Comment by luchunliang** [comment]
please fix the conflicting file

**Comment by emptyOVO** [comment]
> please fix the conflicting file

done

**Review by aloyszhang — DISMISSED** [review]

**Comment by emptyOVO** [comment]
this change will be synchronized with https://github.com/apache/inlong/pull/11011 after #11011  merged

**Comment by emptyOVO** [comment]
> the change of test code will be synchronized with #11011 after #11011 merged

done

**Review by aloyszhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**